packages/test-utils/src/stateChannel.ts

```diff
@@ -1,6 +1,26 @@
 import * as ethers from "ethers";
 import { StructAbiEncoder } from "./structAbiEncoder";
 
+export class App {
+  public static fromContract(contract: ethers.Contract): App {
+    return new App(
+      contract.address,
+      contract.interface.functions.applyAction.sighash,
+      contract.interface.functions.resolve.sighash,
+      contract.interface.functions.turnTaker.sighash,
+      contract.interface.functions.isStateFinal.sighash
+    );
+  }
+
+  constructor(
+    readonly address: string,
+    readonly applyAction: string,
```

Comment:
Consider storing the `ethers.Interface.FunctionDefinition` instead of the sighash string here?

Reply:
That precludes the ability to pass the App object verbatim into a contract call, with no great benefit IMO e.g. `stateChannel.functions.setResolution(app, state, terms)`